File: work/sample-app-scala/src/main/scala/x/HelloSpark.scala
package x

import org.apache.spark.sql.SparkSession
import org.apache.spark.SparkConf
import scala.io.StdIn.readLine
import org.apache.log4j.{Level, Logger}

object HelloSpark {
  def main(args: Array[String]): Unit = {

    Logger.getLogger("org.apache.spark").setLevel(Level.ERROR)

    if (args.length < 1) {
      println("need file(s) to load")
      System.exit(1)
    }

    val conf = new SparkConf().setAppName("HelloSpark")
    //conf.setMaster("spark://localhost:7077") // optional for debugging
    val spark = SparkSession.builder().config(conf).getOrCreate()

    // set the logs to error
    // spark.sparkContext.setLogLevel("ERROR")

    println("### This app UI is available on : " + spark.sparkContext.uiWebUrl.get)

    var file = ""
    for (file <- args) { // looping over files
      val f = spark.read.textFile(file)
      val t1 = System.nanoTime()
      val count = f.count
      val t2 = System.nanoTime()

      println("### %s: count:  %,d ,  time took:  %,f ms".format(file, count, (t2 - t1) / 1e6))
    }

    // HACK : so the 4040 UI stays alive :-)
    println("### Hit enter to terminate the program...:")
    val line = readLine()

    spark.stop() // close the session
  }
}

File: sample-app-java/src/main/java/x/HelloSpark.java
package x;

import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.SparkSession;
import org.apache.log4j.Level;
import org.apache.log4j.Logger;

import java.util.Scanner;

/*
Usage:
// running on local host
$   spark-submit --master local[*]   --class x.HelloSpark target/hello-spark-1.0.jar   /etc/hosts

// submitting to spark master
$   spark-submit --master spark://localhost:7077   --class x.HelloSpark target/hello-spark-1.0.jar   /etc/hosts
 */
 public class HelloSpark {
    
    public static void main(String[] args) {
        Logger.getLogger("org.apache.spark").setLevel(Level.ERROR); // hush logs

        if(args.length < 1) {
            System.out.println ("Need file(s) to load");
            System.exit(1);
        }
        SparkSession spark = SparkSession.builder().appName("Hello Spark").getOrCreate();
        System.out.println("Spark UI available at : " + spark.sparkContext().uiWebUrl().get());

        for(String file : args) {
            Dataset<String> fileDataSet = spark.read().textFile(file);
            
            Long t1 = System.nanoTime();
            Long count = fileDataSet.count();
            Long t2 = System.nanoTime();
            System.out.println (String.format("### %s : count: %,d, Time took: %,d ms",file, count, (t2-t1)/1000000 ));
        }
        // HACK : so the spark UI stays alive :-)
        System.out.println("### Hit enter to terminate the program...:");
        new Scanner(System.in).nextLine();
        spark.stop(); // close the session
    }
}